code/controllers/Processes/timer.dm

```diff
@@ -0,0 +1,77 @@
+var/global/datum/controller/process/timer/PStimer
+
+/datum/controller/process/timer
+	var/list/processing_timers = list()
+	var/list/hashes = list()
+
+/datum/controller/process/timer/setup()
+	name = "timer"
+	schedule_interval = 5 //every 0.5 seconds
+	PStimer = src
+
+/datum/controller/process/timer/statProcess()
+	..()
+	stat(null, "[processing_timers.len] timers")
+
+/datum/controller/process/timer/doWork()
+	if(!processing_timers.len)
+		disabled = 1 //nothing to do, lets stop firing.
+		return
+	for(var/datum/timedevent/event in processing_timers)
+		if(!event.thingToCall || qdeleted(event.thingToCall))
+			qdel(event)
+		if(event.timeToRun <= world.time)
+			spawn(-1)
```

Comment:
Yes yes, I've seen that test and the results, many times---I'm more concerned about how it actually plays out, real world though, with actual pre-existing game code.

For example, in explosion.dm--try running a test with your current setup and take note of the cost+total actual time, then make a change where the only thing changed is that you spawn() instead of using set waitfor = 0 (not for throwing objects; just the explosion itself, for consistency).

I've done this multiple times and every time the set waitfor = 0 explosion is more expensive....any idea why?